react/index.js

```diff
@@ -1,74 +1,38 @@
-import React, { Component } from 'react'
-import { orderFormConsumer, contextPropTypes } from 'vtex.store-resources/OrderFormContext'
-import AddressPage from './components/AddressPage'
-import hoistNonReactStatics from 'hoist-non-react-statics'
-import { head } from 'ramda'
+import React, { useEffect } from 'react'
 import queryString from 'query-string'
-import './global.css'
+import { head, path } from 'ramda'
 
-/**
- * Component responsible for displaying and managing user's address using orderFormContext.
- */
-class AddressManager extends Component {
-  static propTypes = {
-    /* Context used to call address mutation and retrieve the orderForm */
-    orderFormContext: contextPropTypes,
-    /* URL for the store logo */
-    logoUrl: PropTypes.string,
-  }
-
-  handleSelectAddress = () => {
-    this.redirectToReturnURL()
-  }
+import { useAddress, withAddressProvider } from './components/AddressContext'
+import AddressPage from './components/AddressPage'
 
-  componentDidMount() {
-    this.checkIfAddressIsSet()
-  }
+import './global.css'
 
-  componentDidUpdate() {
-    this.checkIfAddressIsSet()
+const redirectToReturnURL = () => {
+  try {
+    const parsedQueryString = queryString.parse(window.location.search)
+    const returnURL = parsedQueryString && parsedQueryString.returnUrl  || ''
+    const cleanUrl = head(returnURL) === '/' ? returnURL : `/${returnURL}`
+    window.location.href = cleanUrl
+  } catch (e) {
+    // Unable to redirect
   }
+}
 
-  checkIfAddressIsSet = () => {
-    const { orderFormContext } = this.props
-    const { shippingData } = orderFormContext.orderForm
-
-    if (shippingData && shippingData.address) {
-      this.redirectToReturnURL()
-    }
-  }
+const AddressManager = props => {
+  const { address } = useAddress()
 
-  redirectToReturnURL = () => {
-    try {
-      const parsedQueryString = queryString.parse(window.location.search)
-      const returnURL = parsedQueryString && parsedQueryString.returnUrl  || ''
-      const cleanUrl = head(returnURL) === '/' ? returnURL : `/${returnURL}`
-      window.location.href = cleanUrl
-    } catch (e) {
-      // Unable to redirect
+  const checkIfAddressIsSet = () => {
+    if (!!path(['orderForm', 'shippingData', 'address'], address)) {
+      redirectToReturnURL()
     }
   }
 
-  render() {
-    const { orderFormContext } = this.props
-    const { shippingData } = orderFormContext.orderForm
-
-    const isLoading = shippingData === undefined
-
-    if (!shippingData || !shippingData.address) {
-      return (
-        <AddressPage
-          loading={isLoading}
-          onSelectAddress={this.handleSelectAddress} />
-      )
-    }
-
-    /** TODO: Add a redirect placeholder (perhaps one of those "If you are not redirected, click here" things)
-     * @author lbebber */
-    return (
-      <AddressPage loading />
-    )
-  }
+  useEffect(() => {
+    checkIfAddressIsSet()
+  })
```

Comment:
I think it might make sense to add `address` here:
```suggestion
  }, [address])
```

Reply:
it definitely does